### PR TITLE
feat: CI job to auto replace assetlists at some interval

### PR DESCRIPTION
**Pull request by deividaspetraitis**
Closes: #7859 

## What is the purpose of the change

This MR introduces a new Github CI Workflow and relevant script to update asset lists periodically _*_, see #7859.

New workflow is using [default](https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#action-behaviour) strategy of peter-evans/create-pull-request action to handle edge cases and avoid spam:
- If there are changes in the source branch, a PR is created.
- If there are no changes in the source branch, no PR is created.
- If there is an existing PR from the source branch to the target branch, it is updated.
- If there is an existing PR from the source branch to the target branch, and there are no changes in the source branch, the PR is closed.

For convenience it is possible to override inputs ( source/target branch ) while manually running the workflow.
```
act workflow_dispatch -j update-asset-lists --secret-file .env --input source-branch=7859
```

A few considerations:

* Please let me know if I should change schedule to something else. 
* Should we adjust labels?
* PR will be created as Draft at this point, should it be non-draft?

## Testing and Verifying

This change can be verified as follows by running workflow manually/locally while leveraging [act](https://github.com/nektos/act):

```
act workflow_dispatch -j update-asset-lists --secret-file .env --input source-branch=7859
```

## Documentation and Release Note
Please let me know if I should document this change somewhere else.

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [X] Code comments?
  - [X] N/A

## Summary by CodeRabbit

- **New Features**
	- Introduced a scheduled workflow to update asset lists and create pull requests automatically.
- **Chores**
	- Added a script to download and update asset lists from specified URLs.

### CODE REVIEW

File: .github/workflows/update-asset-lists.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+# PR is created using DEFAULT strategy of peter-evans/create-pull-request action:
+# - If there are changes in the source branch, a PR is created.
+# - If there are no changes in the source branch, no PR is created.
+# - If there is an existing PR from the source branch to the target branch, it is updated.
+# - If there is an existing PR from the source branch to the target branch, and there are no changes in the source branch, the PR is closed.
+# See more: https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#action-behaviour
+name: Update Asset Lists
+on:
+  schedule:
+    - cron:  '0 0 * * *'
```
Comment (czarcas7ic) [inline .github/workflows/update-asset-lists.yml:19]: Im thinking this should be more like a once a week run max. Honestly I am actually fine with once every 2 weeks. We get assets added regularly and the noise would be a little too much

**Review by czarcas7ic — APPROVED** [review]
Nice work! Good to merge if we reduce how often this runs, I am thinking every 2 weeks is good. The asset list doesn't need to be 100 percent updated all the time, should just tail the current asset list as a reasonable interval

**Comment by ValarDragon** [comment]
Agreed for every 2 weeks

**Comment by deividaspetraitis** [comment]
Thanks for the feedback! I have updated cron expression to run the workflow on the 1st and 15th day of every month at 00:00 UTC. Would that work? Running workflow exactly every two weeks would require some extra tricks since there is no direct cron expression for every 2 weeks.

**Comment by czarcas7ic** [comment]
That works, thank you!

**Pull request merged**